xarray/namedarray/utils.py

```diff
@@ -35,15 +95,15 @@ def module_available(module: str) -> bool:
     return importlib.util.find_spec(module) is not None
 
 
-def is_dask_collection(x: typing.Any) -> bool:
+def is_dask_collection(x: object) -> TypeGuard[DaskCollection]:
     if module_available("dask"):
-        from dask.base import is_dask_collection
+        from dask.typing import DaskCollection
 
-        return is_dask_collection(x)
+        return isinstance(x, DaskCollection)
     return False
```

Comment:
on a related note, i encountered a strange issue while trying to consolidate the two implementations of `is_dask_collection` in the `utils.py` and `pycompat.py` files:https://github.com/pydata/xarray/blob/c9ba2be2690564594a89eb93fb5d5c4ae7a9253c/xarray/namedarray/utils.py#L55-L60 

and 

https://github.com/pydata/xarray/blob/c9ba2be2690564594a89eb93fb5d5c4ae7a9253c/xarray/core/pycompat.py#L89-L94

when using `pint` arrays, i observed different behaviors. Here is an example:

```python
In [22]: from dask.typing import DaskCollection

In [23]: from dask.base import is_dask_collection

In [24]: import xarray as xr, pint_xarray

In [25]: ds =  xr.Dataset({"a": ("x", [0, 1, 2]), "b": ("y", [-3, 5, 1], {"units": "m"})}).pint.quantify(a="s")

In [27]: ds.b
Out[27]: 
<xarray.DataArray 'b' (y: 3)>
<Quantity([-3  5  1], 'meter')>
Dimensions without coordinates: y
```

```python
In [34]: isinstance(ds.b.data, DaskCollection)
Out[34]: True

In [35]: is_dask_collection(ds.b.data)
Out[35]: False
```


@Illviljan, i'm curious to hear your thoughts on how to address this divergence in implementations.

Reply:
> isinstance(x, DaskCollection) 

This doesn't seem right. It just asserts that dask protocols are present on `x`. 

This is always true for xarray objects and pint objects. But these things are technically dask collections if and only if they wrap an underlying dask array.  `is_dask_collection` recognizes this (https://docs.dask.org/en/latest/_modules/dask/base.html#is_dask_collection)

Reply:
just learned that this is a well known issue, 

- https://github.com/pydata/xarray/issues/6578
- https://github.com/dask/dask/issues/9058

Reply:
@dcherian, thank you for chiming in and the pointer to the doc page. i'm going to revert to using `is_dask_collection()` function

Reply:
The function just checked for `.__dask_graph__` in october 2023. Using `isinstance(x, DaskCollection)` correctly narrows down the typing and made sense back then.

Compare the old version: https://github.com/dask/dask/pull/10676

Dask's [definition](https://github.com/dask/dask/blob/daf1f7c73bd2c5d316bbfe2bd8761d05a5128971/dask/typing.py#L106C7-L106C21) of a `DaskCollection` requires `.__dask_keys__` etc. as well which `is_dask_collection` ignores, this is also strange.